readthedocs/proxito/views/mixins.py

```diff
@@ -15,7 +16,7 @@ class ServeDocsMixin:
 
     """Class implementing all the logic to serve a document."""
 
-    def _serve_docs(self, request, final_project, path):
+    def _serve_docs(self, request, final_project, version_slug, path, download=False):
```

Comment:
I wonder if we should default `version_slug` to None, since it isn't needed in the normal case to serve a file. That isn't a great solution either though, since it's a bit confusing to call it. Not really sure what is best.

Reply:
The other option I guess is a top-level `_serve_media` that is similar to `_serve_docs`, which calls into the lower level functions. That way each function could keep only the kwargs it requires.